File: src/java/recursion/Factorial.java
package recursion;

public class Factorial {

    // Not recursive
    public static int factorialNotRecursive(int number) {
        int f = 1;
        for (; number > 1; --number) {
            f *= number;
        }
        return f;
    }

    // Recursive
    public static int factorialRecursive(int number) {
        if (number == 0)
            return 1;
        return (number * factorialRecursive(number - 1));
    }

}

File: src/c++/recursion/factorial.cpp
#include <stdio.h>
  
int factorial(int n) {
    int j, fat;
    fat = 1;

    for (j = 1; j <= n; j++) {
        fat = fat * j;
    }

    return fat;
}

// Recursive
int factorial_recursive(int n) {
    if (n == 0) {
        return 1;
    }
    return n * factorial_recursive(n - 1);
}
